File: frameworks/Java/tapestry/src/main/java/hello/pages/Plaintext.java
package hello.pages;

import org.apache.tapestry5.StreamResponse;
import org.apache.tapestry5.annotations.PageLoaded;
import org.apache.tapestry5.util.TextStreamResponse;

/**
 * Plain-text response test
 */
public class Plaintext
{
  private TextStreamResponse response;
  
  @PageLoaded
  void initializeResponse(){
    response = new TextStreamResponse("text/plain", "Hello, World!");
  }

  StreamResponse onActivate() {
    return response;
  }
}

File: frameworks/TypeScript/typescript-rest/src/controllers/plaintext.ts
import * as express from "express";
import { ContextResponse, GET, Path } from "typescript-rest";

@Path("/plaintext")
export default class Plaintext {
  /**
   * Implements the `Plaintext` test type.
   */

  @GET
  plaintext(@ContextResponse response: express.Response): string {
    response.contentType("text/plain"); // defaults to `text/html` for this function
    return "Hello, World!";
  }
}
